azure-kusto-ingest/source/descriptors.js

```diff
@@ -34,17 +34,17 @@ class FileDescriptor {
     }
 
     async _gzip() {
-        let zipper = zlib.createGzip();
-        let input = fs.createReadStream(this.filePath, { autoClose: true });
-        let output = fs.createWriteStream(this.filePath + ".gz");
+        const zipper = zlib.createGzip();
+        const input = fs.createReadStream(this.filePath, { autoClose: true });
+        const output = fs.createWriteStream(this.filePath + ".gz");
 
         await new Promise((resolve, reject) => {
             input.pipe(zipper).pipe(output)
                 .on("error", (err) => {
                     reject(err);
                 });
             output.once("close", function() {
-                resolve();
+                resolve(null);
```

Comment:
out of interest - why should we add this null?

Reply:
Typescript is strict about number of arguments.
Resolve always accepts 1, but if you don't provide it it's `undefined` in javascript, and fails to compile in typescript.
I sent null instead of undefined because it's more explicit